### PR TITLE
ref: Rename OOM to Watchdog Terminations

### PR DESCRIPTION
**Pull request by kevinrenskers**
Rename OOM to Watchdog Terminations. Basically a pretty simple search and replace without any functional changes.

### CODE REVIEW

**Comment by kevinrenskers** [comment]
The codecov step is resulting in a red cross: `78.42% (-0.03%) compared to 6bb5919`. 

Is that really worth an error? 🤔

**Comment by kevinrenskers** [comment]
Note to self: sentry-docs needs to be updated.

**Comment by kevinrenskers** [comment]
Docs PR: https://github.com/getsentry/sentry-docs/pull/5938. I think we can merge this one? Waiting for approval.

File: Sources/Sentry/Public/SentryOptions.h
Comment (brustolin) [reply Sources/Sentry/Public/SentryOptions.h:141]: I prefer `enableWatchdogTerminationTracking`

**Review by brustolin — APPROVED** [review]
LGTM

Just wait @philipphofmann to vote for the name

**Comment by kevinrenskers** [comment]
I am going to merge this now since it's been waiting for six days and I want to prevent merge conflicts since this touches a bunch of stuff. If we want to change the option name, I can follow up with another PR.

File: Sources/Sentry/Public/SentryOptions.h
Comment (philipphofmann) [reply Sources/Sentry/Public/SentryOptions.h:141]: I would have also preferred `enableWatchdogTerminationTracking`

Comment (kevinrenskers) [reply Sources/Sentry/Public/SentryOptions.h:141]: PR is now open with that change.

Comment (philipphofmann) [reply Sources/Sentry/Public/SentryOptions.h:141]: Thank you @kevinrenskers 😀

**Pull request merged**
